### PR TITLE
Add support for filtering property shapes by group

### PR DESCRIPTION
**Pull request by blushi**
ref: https://github.com/regen-network/regen-registry/issues/455

If `group` is provided as part of the `SHACLValidator` options, it's used to deactivate property shapes without `sh:group` or with `sh:group` different from the given group.

### CODE REVIEW

**Review by haveanicedavid — APPROVED** [review]
Looks good. Not sure if this is relevant, but if this needs to be `.js`, vs `.ts`, we could still [get type safety through JSDoc](https://medium.com/@trukrs/type-safe-javascript-with-jsdoc-7a2a63209b76) (not a priority, just something I was reading about recently)

**Comment by blushi** [comment]
> Looks good. Not sure if this is relevant, but if this needs to be `.js`, vs `.ts`, we could still [get type safety through JSDoc](https://medium.com/@trukrs/type-safe-javascript-with-jsdoc-7a2a63209b76) (not a priority, just something I was reading about recently)

I guess this is a bit out of scope since we're not active maintainers of this repo I have just forked (although we could try to merge this change upstream after we tested it a bit). But I agree some type safety would definitely be nice to have!

**Review by mhagel — APPROVED** [review]

**Pull request merged**
